server/index.ts

```diff
@@ -86,6 +87,57 @@ app
       );
     }
 
+    // Add DNS caching
+    dns.lookup = (() => {
+      const wrappedLookup = function (
+        hostname: string,
+        options:
+          | number
+          | dns.LookupOneOptions
+          | dns.LookupOptions
+          | dns.LookupAllOptions,
+        callback: (
+          err: NodeJS.ErrnoException | null,
+          address: string | dns.LookupAddress[] | undefined,
+          family?: number
+        ) => void
+      ): void {
+        if (typeof options === 'function') {
+          callback = options;
+          options = {};
+        }
+
+        dnsCache
+          .lookup(hostname)
+          .then((result) => {
+            if ((options as dns.LookupOptions).all) {
+              callback(null, [
+                { address: result.address, family: result.family },
+              ]);
+            } else {
+              callback(null, result.address, result.family);
+            }
+          })
+          .catch((error) => {
+            callback(error, undefined, undefined);
+          });
+      } as typeof dns.lookup;
+
+      (wrappedLookup as any).__promisify__ = async function (
+        hostname: string,
+        options?: dns.LookupAllOptions | dns.LookupOneOptions
+      ): Promise<dns.LookupAddress[] | { address: string; family: number }> {
+        const result = await dnsCache.lookup(hostname);
+
+        if (options && 'all' in options && options.all === true) {
+          return [{ address: result.address, family: result.family }];
+        }
+        return { address: result.address, family: result.family };
+      };
+
+      return wrappedLookup;
+    })();
+
```

Comment:
Couldn't you implement this inside the dnsCacheManagerItself? It floods the server a bit.

Reply:
The flood is coming from the debug logging right? Thats to be removed. I have it added rn to sort of test it. Or if not what do you mean?

Reply:
No not that, i meant you could create a function inside `server/utils/dnsCacheManager.ts` that contains all this, because all this code to overwrite the lookup function takes a lot of space inside the startup function of the servers.

Reply:
Ah. And then call it here?

Reply:
yes

Reply:
Will clean it up (also remove the debug spams) once i get this working properly